catalog/core/catalog-core-commands/src/test/java/org/codice/ddf/commands/catalog/RemoveCommandTest.java

```diff
@@ -99,33 +165,99 @@ protected SolrCacheMBean getCacheProxy() {
      */
     @Test
     public void testNullList() throws Exception {
-        final SolrCacheMBean mbean = mock(SolrCacheMBean.class);
-        RemoveCommand removeCommand = new RemoveCommand() {
-            @Override
-            protected SolrCacheMBean getCacheProxy() {
-                return mbean;
-            }
-        };
-
         removeCommand.ids = null;
-
         removeCommand.executeWithSubject();
-
         assertThat(consoleOutput.getOutput(), containsString("Nothing to remove."));
     }
 
-    private java.util.List<Metacard> getMetacardList(int amount) {
-        List<Metacard> metacards = new ArrayList<>();
+    private void setQueryAndDeleteResponseMocks(int... numResultsPerQuery) {
+        setQueryResponseMockReturn(numResultsPerQuery);
+        setDeleteResponseMockReturn(numResultsPerQuery);
+    }
+
+    /*
+     * Given the internal implementation of RemoveCommand:executeRemoveFromStore(),
+     * queryResponse.getResults() needs to be called three times per catalog framework
+     * query. For this reason, three mock responses are added for each argument.
+     *
+     * @param numResultsPerQueryList
+     */
+    private void setQueryResponseMockReturn(int... numResultsPerQueryList) {
+
+        List<List<Result>> queryResponseMockList = new ArrayList<>();
+
+        for (int numResultsForQuery : numResultsPerQueryList) {
+
+            List<Result> resultList = populateResultList(numResultsForQuery);
+            queryResponseMockList.add(resultList);
+
+            // add two copies of first result list as implementation of RemoveAllCommand
+            // calls queryResponse.getResults() three times per catalog framework query
```

Comment:
This command is not `removeall` and does not do that. I don't think the first result should be added three times

Reply:
Oops, I need to have Weser remind me why it's necessary to add 3 times in `removeall`

Reply:
This seems like the tests are way too tightly coupled to the implementation 😞

Reply:
Update: The ResultIterable also will call 3 `getResults`, so it's unfortunate. Maybe there's a better way to do this without knowing all of that